armi/reactor/reactorParameters.py

```diff
@@ -770,3 +770,29 @@ def defineCoreParameters():
         )
 
     return pDefs
+
+
+def makeParametersReadOnly(r):
+    """Convert all the paramters in a Reactor to read-only.
+
+    This method is pretty simple. It goes through all the children of a Reactor object,
+    recursively, and converts the parameters to read-only mode.
+
+    Parameters
+    ----------
+    r : Reactor
+        Full reactor object, to be modified.
+
+    Notes
+    -----
+    Once you make one Reactor read-only, you cannot make it writeable again.
+    """
+    r.p.readOnly = True
```

Comment:
Hey @john-science I had a comment on this related to assuming this is the structure of the rector. Is there a better way to do a recursive get all children from the top level? Right now this could be fragile

Reply:
Wait, one sec.

Reply:
done